File: Priority_Queue/3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long, vector<long>, greater<long>> pq;
        for(int num : nums) pq.push((long) num);
        int count = 0;

        while(pq.size() > 1 && pq.top() < k){
            long a = pq.top(); pq.pop();
            long b = pq.top(); pq.pop();

            pq.push(a * 2 + b);
            count++;
        }

        return count;
    }
};

File: Priority_Queue/3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for(int num:nums) pq.offer((long) num);
        int count = 0;

        while(pq.size() > 1 && pq.peek() < k){
            long a = pq.poll();
            long b = pq.poll();

            pq.offer(a*2 + b);
            count++;
        }

        return count;
    }
}